### PR TITLE
feat: Implement options for EXPIRE command

### PR DESCRIPTION
**Pull request by UkuLoskit**
Add NX, XX, LT, GT options to EXPIRE, EXPIREAT, PEXPIRE and PEXPIREAT.

https://github.com/dragonflydb/dragonfly/issues/842

### CODE REVIEW

File: src/server/db_slice.cc [hunk 1]
```diff
@@ -732,9 +733,22 @@ OpResult<int64_t> DbSlice::UpdateExpire(const Context& cntx, PrimeIterator prime
...
   } else if (IsValid(expire_it) && !params.persist) {
+    auto current = ExpireTime(expire_it);
+    if (params.expire_options & ExpireFlags::EXPIRE_NX) {
+      return OpStatus::SKIPPED;
+    }
+    if (params.expire_options & ExpireFlags::EXPIRE_LT && current <= abs_msec) {
+      return OpStatus::SKIPPED;
+    } else if (params.expire_options & ExpireFlags::EXPIRE_GT && current >= abs_msec) {
+      return OpStatus::SKIPPED;
+    }
```
Comment (dranikpg) [inline src/server/db_slice.cc:744]: Please add parentheses around the bit ops

File: src/server/generic_family.cc [hunk 2]
```diff
@@ -734,6 +734,43 @@ void GenericFamily::Persist(CmdArgList args, ConnectionContext* cntx) {
...
+      flags |= ExpireFlags::EXPIRE_LT;
+      lt = true;
+    } else {
+      (*cntx)->SendError(absl::StrCat("Unsupported option: ", ToSV(arg)));
+      return nullopt;
+    }
+  }
+  if ((nx && xx) || (nx && gt) || (nx && lt)) {
+    (*cntx)->SendError("NX and XX, GT or LT options at the same time are not compatible");
+    return nullopt;
```
Comment (dranikpg) [inline src/server/generic_family.cc:765]: you can check `(flags & NX) && (flags & ~NX)`, which means that both NX and some other option are set, its quite compact, so you don't need the bool variables

File: src/server/generic_family.cc [hunk 3]
```diff
@@ -734,6 +734,43 @@ void GenericFamily::Persist(CmdArgList args, ConnectionContext* cntx) {
     (*cntx)->SendLong(0);
 }
 
+std::optional<int32_t> parseExpireOptions(const CmdArgList args, ConnectionContext* cntx) {
+  int32_t flags = ExpireFlags::EXPIRE_ALWAYS;
```
Comment (dranikpg) [inline src/server/generic_family.cc:738]: Please start with a captial letter, you can also name it `ParseExpireOptionsOrReply` for clarity

**Review by dranikpg — APPROVED** [review]
Hi. Great work 🔥 Highlighted some really minor issues

**Comment by UkuLoskit** [comment]
@dranikpg thanks for the quick feedback. I have addressed the comments.

**Review by dranikpg — APPROVED** [review]

**Comment by dranikpg** [comment]
Thanks for your contribution! As always superb 🎩

**Pull request merged**